### ISSUE TITLE
Prow "trigger" plugin not working?

### ISSUE THREAD

**Issue by bobcatfish** [tektoncd/plumbing#582]
# Expected Behavior

1. The status of all prow checks on a PR should be accurate
2. It should be possible to re-trigger the running of prow jobs with `/test` and `/retest`

# Actual Behavior

Checks on some PRs are stuck and cannot be triggered with `/test` or `/retest`

I think that they got stuck in a pending state for <some reason> but now that the  `/test` and `/retest` commands are not working because the prow "trigger" plugin is not working

I've tried using `/retest` repeatedly on https://github.com/tektoncd/pipeline/pull/3271 and it is doing nothing.

I've also tried restarting hook and @afrittoli tried restarting plank. Another candidate for a restart is crier but it seems the jobs arent even being started.

## Stuck PRs

1. https://github.com/tektoncd/pipeline/pull/3269

pull-tekton-pipeline-unit-tests is stuck pending:
![image](https://user-images.githubusercontent.com/432502/93826563-7f740e80-fc35-11ea-9913-434e27c47ae6.png)

If you follow the details link, it appears to have succeeded: https://tekton-releases.appspot.com/build/tekton-prow/pr-logs/pull/tektoncd_pipeline/3269/pull-tekton-pipeline-unit-tests/1308069547818881025/

Neither /test nor /retest have any effect.

2. https://github.com/tektoncd/pipeline/pull/3144

pull-tekton-pipeline-unit-tests WAS stuck on this PR (in the same state iirc: it claimed to be pending but has succeeded every time it's been run https://tekton-releases.appspot.com/builds/tekton-prow/pr-logs/pull/tektoncd_pipeline/3144/pull-tekton-pipeline-unit-tests/) but updating the commit (i.e. force pushing a new commit) seems to have finally triggered it

(maybe some component is tracking commit IDs and thinks it doesn't need to know about the stuck ones?)

3. https://github.com/tektoncd/pipeline/pull/3168

pull-tekton-pipeline-go-coverage and pull-tekton-pipeline-unit-tests are both stuck. neither /test nor /retest have any effect.

![image](https://user-images.githubusercontent.com/432502/93826828-1345da80-fc36-11ea-86be-a73b97546f88.png)

# Additional Info

I tried to investigate https://github.com/tektoncd/pipeline/pull/3168

1. Made /retest comment (https://github.com/tektoncd/pipeline/pull/3168#issuecomment-696407447)
2. Looked at recent deliveries to the webhook to verify that the event was delivered (https://github.com/organizations/tektoncd/settings/hooks/92587262)

![image](https://user-images.githubusercontent.com/432502/93827446-48066180-fc37-11ea-9bdd-d3e7f32902d2.png)

Seems kinda weird that the request URL isn't complete (http://34.95.121.238/hook) but that seems to be the case for all deliveries?

3. Based on [prow architecture diagram](https://github.com/kubernetes/test-infra/blob/master/prow/docs/architecture.png) hook should be invoking the "trigger" plugin for this:

http://34.95.121.238/command-help
![image](https://user-images.githubusercontent.com/432502/93827600-90258400-fc37-11ea-863f-c0c8e2e83174.png)

Looked at logs for hook, saw the same error @afrittoli pointed out, maybe this is stopping hook from invoking any plugins?

```
"Post http://needs-rebase: dial tcp: lookup needs-rebase on 10.55.240.10:53: no such host" 
```

Looking at the code at this point in time it doesn't seem this not working should block built in plugins from working: https://github.com/kubernetes/test-infra/blob/e3f7b98532519da4de88d65739e28802b426b265/prow/hook/server.go#L200

And the /retest command seems to have worked as recently as 7 hours ago: https://github.com/tektoncd/pipeline/pull/3266

Looking at the monitoring dashboard it looks like there might be a memory leak somewhere:
![image](https://user-images.githubusercontent.com/432502/93828653-c8c65d00-fc39-11ea-8f37-9cf335a019da.png)

**Comment by bobcatfish** [comment]
Also did not work in https://github.com/tektoncd/dashboard/pull/1711

**Comment by vdemeester** [comment]
Note that `/retest` usually only re-run failed tests — in the case of "blocked" ones, it won't re-run most likely (if they are actually successful but didn't push their status back)

**Comment by vdemeester** [comment]
```
default/hook-774978fc77-g5xsr[hook]: {"author":"vdemeester","component":"hook","event-GUID":"b8e16400-fc9f-11ea-909f-920d6379c3bd","event-type":"issue_comment","file":"prow/hook/events.go:293","func":"k8s.io/test-infra/prow/hook.(*Server).handleIssueCommentEvent","level":"info","msg":"Issue comment created.","org":"tektoncd","pr":3269,"repo":"pipeline","time":"2020-09-22T06:49:12Z","url":"https://github.com/tektoncd/pipeline/pull/3269#issuecomment-696540847"}
default/hook-774978fc77-g5xsr[hook]: {"component":"hook","error":"Post http://needs-rebase: dial tcp: lookup needs-rebase on 10.55.240.10:53: no such host","event-GUID":"b8e16400-fc9f-11ea-909f-920d6379c3bd","event-type":"issue_comment","external-plugin":"needs-rebase","file":"prow/hook/server.go:200","func":"k8s.io/test-infra/prow/hook.(*Server).demuxExternal.func1","level":"error","msg":"Error dispatching event to external plugin.","time":"2020-09-22T06:49:15Z"}
```

**Comment by vdemeester** [comment]
So few things:
- It is possible to restart a job from `kubectl` (kubectl create -f "https://prow.tekton.dev/rerun?prowjob=3d038d93-fc21-11ea-9847-7e6f0dcb7dc3" for tektoncd/pipeline#3269 that unblocked it)
- If the jobs are completely gone (like on tektoncd/pipeline#3168), I don't see any other way but to force-push or use `/test all`
- `/retest` only retest failed jobs, if you want to *force* re-run all jobs, `/test all` is your friend
- the `needs-rebase` problem is fixed, but wasn't related to this

I think it works *as expected*, there is some weird case where.. prow looses jobs of opened PRs. And in a gist: **if you want to re-run all jobs, use `/test all`**, it force the run of all, and doesn't try to be smart.

**Comment by vdemeester** [comment]
(I think we can close this issue btw :stuck_out_tongue: )

**Comment by bobcatfish** [comment]
> /retest only retest failed jobs, if you want to force re-run all jobs, /test all is your friend

ah kk!

@vdemeester do you want to keep this open to track the stuck PRs as well or do you prefer to open a new issue for that? (unless we've resolved it?)

**Comment by vdemeester** [comment]
> @vdemeester do you want to keep this open to track the stuck PRs as well or do you prefer to open a new issue for that? (unless we've resolved it?)

yeah, I'm fine keeping it opened, but stuck PRs will happen from time to time, we can "relatively" easily manually unblock them (all the pipeline stuck ones are not stuck anymore, …)

**Comment by bobcatfish** [comment]
i dont think we've seen this happen for a loooong while
